# All of hunks of this patch are picked from ungoogled-chromium's remove-uneeded-ui.patch

# Removes Safety Check section and Privacy Guide from the settings page
--- a/chrome/browser/resources/settings/privacy_page/privacy_page.html
+++ b/chrome/browser/resources/settings/privacy_page/privacy_page.html
@@ -23,15 +23,6 @@
         label="$i18n{clearBrowsingData}"
         sub-label="$i18n{clearBrowsingDataDescription}"
         on-click="onClearBrowsingDataClick_"></cr-link-row>
-    <template is="dom-if" if="[[isPrivacyGuideAvailable]]">
-      <cr-link-row id="privacyGuideLinkRow" class="hr"
-          start-icon="privacy20:signpost"
-          label="$i18n{privacyGuideLabel}"
-          sub-label="$i18n{privacyGuideSublabel}"
-          on-click="onPrivacyGuideClick_"
-          role-description="$i18n{subpageArrowRoleDescription}">
-      </cr-link-row>
-    </template>
     <cr-link-row id="thirdPartyCookiesLinkRow"
         start-icon="privacy:cookie"
         class="hr" label="$i18n{thirdPartyCookiesLinkRowLabel}"
--- a/chrome/browser/resources/settings/privacy_page/privacy_page_index.html
+++ b/chrome/browser/resources/settings/privacy_page/privacy_page_index.html
@@ -14,14 +14,6 @@
 <cr-view-manager id="viewManager" class="cr-centered-card-container"
     show-all$="[[shouldShowAll]]">
 
-  <template is="dom-if" if="[[isPrivacyGuideAvailable]]" restamp>
-    <settings-section slot="view" id="privacyGuidePromo" page-title=""
-        hidden$="[[!showPrivacyGuidePromo_]]">
-      <settings-privacy-guide-promo prefs="{{prefs}}">
-      </settings-privacy-guide-promo>
-    </settings-section>
-  </template>
-
   <template is="dom-if" if="[[showPage_(pageVisibility_.safetyHub)]]">
     <template is="dom-if" if="[[renderView_(
         routes_.PRIVACY, currentRoute, inSearchMode)]]" update-when-false>
--- a/chrome/browser/resources/settings/privacy_page/privacy_page_index.ts
+++ b/chrome/browser/resources/settings/privacy_page/privacy_page_index.ts
@@ -7,7 +7,6 @@ import '/shared/settings/prefs/prefs.js'
 import '../safety_hub/safety_hub_entry_point.js';
 import '../settings_page/settings_section.js';
 import '../settings_shared.css.js';
-import './privacy_guide/privacy_guide_promo.js';
 import './privacy_page.js';
 
 import {PrefsMixin} from '/shared/settings/prefs/prefs_mixin.js';
@@ -235,13 +234,6 @@ export class SettingsPrivacyPageIndexEle
   private getDefaultViews_(): string[] {
     const defaultViews = ['privacy'];
 
-    if (this.isPrivacyGuideAvailable) {
-      defaultViews.push('privacyGuidePromo');
-    }
-
-    if (this.showPage_(this.pageVisibility_.safetyHub)) {
-      defaultViews.push('safetyHubEntryPoint');
-    }
 
     return defaultViews;
   }
@@ -373,19 +365,7 @@ export class SettingsPrivacyPageIndexEle
   }
 
   private updatePrivacyGuidePromoVisibility_() {
-    if (!this.isPrivacyGuideAvailable || this.prefs === undefined ||
-        this.getPref('privacy_guide.viewed').value ||
-        this.privacyGuideBrowserProxy_.getPromoImpressionCount() >=
-            MAX_PRIVACY_GUIDE_PROMO_IMPRESSION ||
-        this.currentRoute !== routes.PRIVACY) {
       this.showPrivacyGuidePromo_ = false;
-      return;
-    }
-    this.showPrivacyGuidePromo_ = true;
-    if (!this.privacyGuidePromoWasShown_) {
-      this.privacyGuideBrowserProxy_.incrementPromoImpressionCount();
-      this.privacyGuidePromoWasShown_ = true;
-    }
   }
 }
 
# Remove the password leak detection toggle on the security settings page
# Remove the Advanced Protection Program link on the security settings page
--- a/chrome/browser/resources/settings/privacy_page/security/security_page.html
+++ b/chrome/browser/resources/settings/privacy_page/security/security_page.html
@@ -115,13 +115,6 @@
     <div class="cr-row first">
       <h2 class="cr-title-text">$i18n{advancedPageTitle}</h2>
     </div>
-    <settings-toggle-button id="passwordsLeakToggle"
-          label="$i18n{passwordsLeakDetectionLabel}"
-          pref="{{prefs.generated.password_leak_detection}}"
-          sub-label="[[getPasswordsLeakToggleSubLabel_(
-                        prefs.profile.password_manager_leak_detection.*,
-                        prefs.generated.password_leak_detection.*)]]">
-    </settings-toggle-button>
     <template is="dom-if" if="[[!enableHttpsFirstModeNewSettings_]]" restamp>
       <settings-toggle-button id="httpsOnlyModeToggle"
           pref="{{prefs.generated.https_first_mode_enabled}}"
@@ -172,10 +165,4 @@
         on-click="onManageCertificatesClick_" external>
     </cr-link-row>
 
-    <cr-link-row id="advancedProtectionProgramLink" class="hr"
-        label="$i18n{advancedProtectionProgramTitle}"
-        sub-label="$i18n{advancedProtectionProgramDesc}"
-        on-click="onAdvancedProtectionProgramLinkClick_"
-        external>
-    </cr-link-row>
   </settings-subpage>
